src/main/java/com/autotune/experimentManager/core/ExperimentTrialHandler.java

```diff
@@ -159,32 +159,29 @@ public void startExperimentTrials() {
                 IntStream.rangeClosed(1, numberOFIterations).forEach(
                     i -> {
                         deploymentHandler.initiateDeploy();
+
                         //Check if deployment is ready
-                        int deploymentIsReadyWithinMinute = EMConstants.TimeConv.DEPLOYMENT_IS_READY_WITHIN_MINUTE;
-                        long millisMinutes = TimeUnit.MINUTES.toMillis(deploymentIsReadyWithinMinute);
-                        int count = (int)millisMinutes/EMConstants.TimeConv.DEPLOYMENT_CHECK_INTERVAL_IF_READY_MILLIS;
-                        for (int j = count; j > 0 && !deploymentHandler.isDeploymentReady(); j--) {   //Parameterize sleep duration hardcoded for 120
-                            try {
-                                LOGGER.debug("Still deployment is not ready for ExpName {} trail No {}", this.experimentTrial.getExperimentName(), tracker);
-                                Thread.sleep(EMConstants.TimeConv.DEPLOYMENT_CHECK_INTERVAL_IF_READY_MILLIS);
-                            } catch (InterruptedException e) {
-                                e.printStackTrace();
-                            }
+                        EMUtil.DeploymentReadinessStatus deploymentReadinessStatus = deploymentHandler.isDeploymentReady();
```

Comment:
Need to check if deployment is needed and only then call initiateDeploy if needed. In the monitoring only usecase, we might not need to do any deployment at all

Reply:
This also needs to be further tied into `input.json`. The input.json validation phase needs to set a flag that deployment is needed for this trial. That flag can be used here to determine if `initiateDeploy` needs to be called.

Reply:
Raised #502 to make the flags check, limiting this PR scope to abstract out the deployment ready part